Fix UB in benchmark

hebasto opened the pull request:
From https://en.cppreference.com/w/c/language/operator_arithmetic.html:
> The behavior is undefined if _`rhs`_ is negative or is greater or equal the number of bits in the promoted _`lhs`_.

Steps to reproduce on Ubuntu 25.04 using the master branch @ ea8f66b1eabb0388d221daaa4706120a5083da0c:
```
$ ./autogen.sh
$ ./configure --enable-benchmark CXX=clang++-20 CXXFLAGS="-O0"
$ make bench -j $(nproc) bench
$ ./bench
<snip>
recover[ms]	 63	 260.72278	         -	  62.21269	
create[ns]	 63	  79.66483	         -	  17.74962	
/usr/lib/gcc/x86_64-linux-gnu/15/../../../../include/c++/15/bits/uniform_int_dist.h:108: std::uniform_int_distribution<unsigned long>::param_type::param_type(_IntType, _IntType) [_IntType = unsigned long]: Assertion '_M_a <= _M_b' failed.
Aborted (core dumped)
```

sipa commented on src/bench.cpp:
```diff
@@ -42,7 +43,13 @@ int main(int argc, char** argv) {
             std::vector<uint64_t> roots(2 * syndromes);
             std::random_device rng;
-            std::uniform_int_distribution<uint64_t> dist(1, (uint64_t(1) << bits) - 1);
+            uint64_t upper_bound;
```
There is a simpler way: 

```c++
auto upper_bound = std::numeric_limits<uint64_t>::max() >> (64 - bits);
```

hebasto replied:
Thanks! Taken.

hebasto committed:
Fix UB in benchmark

From https://en.cppreference.com/w/c/language/operator_arithmetic.html:
> The behavior is undefined if rhs is negative or is greater or equal
> the number of bits in the promoted lhs.

Co-authored-by: Pieter Wuille <pieter@wuille.net>

hebasto committed:
ci: Enable benchmarks for all Windows jobs

maflcko reviewed (approved):
lgtm
